packages/grid/x-data-grid-pro/src/hooks/features/columnReorder/useGridColumnReorder.tsx

```diff
@@ -97,6 +98,69 @@ export const useGridColumnReorder = (
       });
 
       originColumnIndex.current = apiRef.current.getColumnIndex(params.field, false);
+
+      const draggingColumnGroupPath = apiRef.current.getColumnGroupPath(params.field);
+
+      const visibleColumnIndex = apiRef.current.getColumnIndex(params.field, true);
+      const visibleColumns = apiRef.current.getVisibleColumns();
+      const groupsLookup = apiRef.current.getAllGroupDetails();
+
+      // The limitingGroupId is the id of the group from which the dragged column should not escape
+      let limitingGroupId: string | null = null;
+
+      draggingColumnGroupPath.forEach((groupId) => {
```

Comment:
Nice! We need to implement the same (or similar) solution for the row reordering when a row is inside a group.